pkg/azure/aks/machine_store.go

```diff
@@ -209,15 +209,41 @@ func (m *MachineStore) PopulateMachineStore(ctx context.Context) error {
 			continue
 		}
 
-		vmInfo, err := m.getVmInfoFromResourceGroup(ctx, rgName, clusterName)
-		if err != nil {
-			return err
-		}
+		eg.Go(func() error {
+			vmssMap, err := m.getVMSSInfoFromResourceGroup(nestedCtx, rgName, clusterName)
+			if err != nil {
+				return err
+			}
 
-		for name, info := range vmInfo {
-			m.MachineMap[name] = info
-		}
+			for vmssName, vmssInfo := range vmssMap {
+				eg.Go(func() error {
+					vmssPriority := getMachineScaleSetPriority(vmssInfo)
+					vmssOperationSystem := getMachineScaleSetOperatingSystem(vmssInfo)
+
+					vmssVmInfo, err := m.getVmInfoFromVmss(nestedCtx, rgName, vmssName, vmssPriority, vmssOperationSystem)
+					if err != nil {
+						return err
+					}
+
+					vmInfoLock.Lock()
+					for vmName, vmInfo := range vmssVmInfo {
+						vmInfoMap[vmName] = vmInfo
+					}
+					vmInfoLock.Unlock()
+					return nil
+				})
+			}
+			return nil
+		})
+	}
```

Comment:
IMO I would add some way of restricting the amount of go routines that can be generated. You can use [Setlimit](https://pkg.go.dev/golang.org/x/sync/errgroup#Group.SetLimit), and I'd create a const for the max number of goroutines. Maybe 10?

Reply:
Yup, added a 10 limit.